src/compiler/checker.ts

```diff
@@ -2060,10 +2061,10 @@ namespace ts {
             let symbol: Symbol;
             if (name.kind === SyntaxKind.Identifier) {
                 const message = meaning === namespaceMeaning ? Diagnostics.Cannot_find_namespace_0 : Diagnostics.Cannot_find_name_0;
-
-                symbol = resolveName(location || name, name.escapedText, meaning, ignoreErrors ? undefined : message, name, /*isUse*/ true);
+                const secondarySymbol = resolveEntityNameFromJSPrototype(name, meaning);

```

Comment:
consider adding `isInJavascriptFile` just for clarity.

Reply:
And maybe a comment that says "secondarySymbol is undefined except for type references in jsdoc on javascript prototype assignments"

Or something simpler than that. Hopefully.